frontend/src/pages/map/layers/BasemapControl.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMap } from 'react-leaflet'
+import L from 'leaflet'
+import { useEffect } from 'react'
+import 'leaflet-switch-basemap'
+import 'leaflet-switch-basemap/src/L.switchBasemap.css'
+
+declare module 'leaflet' {
+  class basemapsSwitcher {
+    constructor(basemaps: any[], options?: any)
+    addTo(map: any): this
+  }
+}
+
+// TODO: Replace with a more permanent key if we choose to use a MapTiler basemap.
+const params = new URLSearchParams(window.location.search)
+const urlKey = params.get('vite_maptiler_key')
+// If no key in the URL, use the one in the .env file.
+const key = urlKey || import.meta.env.VITE_MAPTILER_KEY
```

Comment:
we could use env file to get the env vars, ex: 
```
import` { env } from '@/env'
// Feature flag for turning OpenStreetMap tiles gray
  const osmGrayscale = env.VITE_OSM_GRAYSCALE_FLAG === 'true'
```
https://github.com/bcgov/nr-epd-organics-info/blob/main/frontend/src/pages/map/MapView.tsx#L31

Reply:
Good idea Ohm.

Reply:
Ended up removing the basemaps that needed this token...  So no longer an issue.